pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentStreamObserver.java

```diff
@@ -151,22 +152,22 @@ public void onNext(Mailbox.MailboxContent mailboxContent) {
 
   @Override
   public void onError(Throwable e) {
+    initError(e);
+  }
+
+  @Override
+  public void onCompleted() {
+    _isCompleted.set(true);
+  }
+
+  private void initError(Throwable e) {
     try {
       _errorLock.writeLock().lock();
       _errorContent = createErrorContent(e);
-      _gotMailCallback.accept(_mailboxId);
-      // TODO: close the stream.
-      throw new RuntimeException(e);
-    } catch (IOException ioe) {
-      throw new RuntimeException("Unable to encode exception for cascade reporting: " + e, ioe);
+    } catch (IOException ioException) {
+      LOGGER.error("Unexpected error", ioException);
     } finally {
       _errorLock.writeLock().unlock();
     }
   }
-
-  @Override
-  public void onCompleted() {
-    _isCompleted.set(true);
-    _responseObserver.onCompleted();
```

Comment:
This cannot be removed. it is an ack sent back to close grpc

Reply:
+1